File: v-loafer/loafer.java/src/main/java/org/example/loafer/model/common/BaseModel.java
package org.example.loafer.model.common;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import lombok.Data;

import java.time.LocalDateTime;

@Data
public abstract class BaseModel {
    @TableId
    private Long id;
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
    private Long createdBy;
}

File: v-cabinet/cabinet.kt/src/main/kotlin/org/example/model/common/BaseModel.kt
package org.example.model.common

import org.babyfish.jimmer.sql.MappedSuperclass
import java.time.Instant

@MappedSuperclass
interface BaseModel {
    val id: Long?
    val createdAt: Instant?
    val updatedAt: Instant?
    val deletedAt: Instant?
}